Navigations

SRWieZ opened the pull request:
List of modifications
- [x] sticky header navigation
- [x] highlight current on right side navigation
- [x] Show ToC persistently on the right-hand side
- [x] Add pinkary / Linkedin / open collective icons
- [x] Check dark mode
- [x] Mobile menu
- [x] Add a copy button to torchlight

simonhamp commented:
Can you add some screenshots to the PR? I'm out at the moment so I can't run this branch

SRWieZ commented:
Done for the day 😉

A before => After

<img width="45%" alt="Capture d’écran 2024-09-14 à 17 53 23" src="https://github.com/user-attachments/assets/ef309483-b9d4-4aac-88c2-1745845f7269">
<img width="45%" alt="Capture d’écran 2024-09-14 à 17 53 28" src="https://github.com/user-attachments/assets/d0d86f8a-91ba-4ea8-b144-0e1d959b7daa">

SRWieZ commented:
Darkmode

<img width="45%" alt="Capture d’écran 2024-09-15 à 17 01 43" src="https://github.com/user-attachments/assets/05d61b15-b64a-4dc5-a45e-c42159db1d20">
<img width="45%" alt="Capture d’écran 2024-09-15 à 17 01 46" src="https://github.com/user-attachments/assets/f3927539-f722-41fb-99de-8b9df1c1c9ec">



Darkmode  mobile

<img width="45%" alt="Capture d’écran 2024-09-15 à 17 02 02" src="https://github.com/user-attachments/assets/5ae0439a-ceb5-4c8c-82c2-14bf0473bcc0">
<img width="45%" alt="Capture d’écran 2024-09-15 à 17 02 06" src="https://github.com/user-attachments/assets/3ef96e40-f966-4260-8c4c-c1288811889e">


Mobile menu

<img width="45%" alt="Capture d’écran 2024-09-15 à 17 02 19" src="https://github.com/user-attachments/assets/827b4c54-57a6-46e8-8ac1-e97916bec1c2">
<img width="45%" alt="Capture d’écran 2024-09-15 à 17 02 21" src="https://github.com/user-attachments/assets/0a36d2af-2e57-4237-9cb0-7f17080664c8">

SRWieZ commented:
It's ready for review @simonhamp!

I'm sorry for the scope of this PR. It was meant to stay on the navigation side, but when I started working on the ToC, I needed to change the layout and other things. After that, it did get out of hand; I added a copy button on the code block and rewrote the responsive menu.

The best way to review is to check out this branch and look side by side with the production version.

I tried to keep things mostly in the same place.

I did a self-review today, and I think it's ready to be merged as it is. 

I can make changes in other PR if needed. In future PRs, I would like to:
-  Uniformize Events documentation; it's different on each page (MenuBar, PowerMonitor, Windows)
-  Rename some things (why is the header nav called banner? 😄)
-  Extract some components maybe (logo list)

simonhamp commented:
This is really great! Thanks so much 🙏🏼
